### PR TITLE
Add additional tests for bufwrite.c

### PR DESCRIPTION
**Pull request by yegappan**

### CODE REVIEW

File: src/testdir/test_writefile.vim [hunk 1]
```diff
@@ -58,7 +58,16 @@ func Test_writefile_fails_conversion()
   call assert_fails('write ++enc=cp932', 'E513:')
   call assert_equal(contents, readfile('Xfile'))
 
+  " With 'backupcopy' set, if there is a conversion error, the backup file is
+  " still creaetd.
```
Comment (dpelle) [inline src/testdir/test_writefile.vim]: typo: creatd → created

File: src/testdir/test_writefile.vim [hunk 2]
```diff
@@ -257,6 +269,15 @@ func Test_write_errors()
   let long_fname = repeat('n', 5000)
   call assert_fails('exe "w " .. long_fname', 'E75:')
   call assert_fails('call writefile([], long_fname)', 'E482:')
+
+  " Test for writing to a block device on Unix-like systems
+  if has('unix') && getfperm('/dev/loop0') != ''
```
Comment (dpelle) [inline src/testdir/test_writefile.vim:284]: Maybe safer if we also check `getftype('/dev/loop0') ==  'bdev'`?

I also wonder what happens if we run this test as root.

Comment (yegappan) [reply src/testdir/test_writefile.vim:284]: I ran the test as root and the test passed.

I see some other test failure in FreeBSD (Cirrus CI). I am not able to reproduce this failure in my local FreeBSD system.

**Comment by brammool** [comment]
It goes appear to fail consistently on FreeBSD.  Perhaps it is running as root and something else?

**Comment by vim-ml** [comment]
Hi Bram,

On Mon, May 10, 2021 at 2:19 AM Bram Moolenaar ***@***.***>
wrote:

> It goes appear to fail consistently on FreeBSD. Perhaps it is running as
> root and something else?
>
>
> The failures are caused by the 'backupskip' option. The Cirrus CI tests
are run from the /tmp directory and backups are not created for files
in the /tmp directory (because of the backupskip option).

- Yegappan

**Pull request closed without merging**